scrapyd_k8s/api.py

```diff
@@ -137,6 +143,41 @@ def enable_authentication(app, config_username, config_password):
     app.config["BASIC_AUTH_FORCE"] = True
     return basic_auth
 
+def watch_pods(namespace):
+    num_lines_to_check = scrapyd_config.get('num_lines_to_check')
+    w = watch.Watch()
+    v1 = client.CoreV1Api()
+    for event in w.stream(v1.list_namespaced_pod, namespace=namespace):
+        pod = event['object']
+        # check the labels in the docs
+        if pod.status.phase == 'Running' and pod.metadata.labels.get("org.scrapy.job_id"):
+            thread_name = "%s_%s" % (namespace, pod.metadata.name)
+            if (thread_name in watcher_threads
+                    and watcher_threads[thread_name] is not None
+                    and watcher_threads[thread_name].is_alive()):
+                pass
+            else:
+                watcher_threads[thread_name] = threading.Thread(
+                    target=stream_logs,
+                    kwargs={
+                        'job_name': pod.metadata.name,
+                        'namespace': namespace,
+                        'num_lines_to_check': num_lines_to_check
+                    }
+                )
+                watcher_threads[thread_name].start()
+        elif pod.status.phase == 'Succeeded' and pod.metadata.labels.get("org.scrapy.job_id"):
+            log_filename = make_log_filename_for_job(pod.metadata.name)
+            if os.path.isfile(log_filename):
+                if object_storage_provider.is_local_file_uploaded(log_filename):
+                    print("file already exists")
+                else:
+                    object_storage_provider.upload_file(log_filename)
+            else:
+                print("logfile not found %s", pod.metadata.name)
+        else:
+            print("other type " + event["type"] + " " + pod.metadata.name + " - " + pod.status.phase)
```

Comment:
Can we move this piece of code to the log handling file? I'd like to keep the `api.py` file as clean as possible. Then the job log handling functionality kind of 'uses' the watch functionality provided by the main api program.